### PR TITLE
Add cfg for RTD

### PR DESCRIPTION
**Pull request by trey-stafford**
Most of this is pulled directly from the scientific python template / guide.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Sphinx documentation
+docs/_build/
+docs/api/
```
Comment (trey-stafford) [inline .gitignore:3]: My thinking is that the api docs should not be in source control. They're based on the source code and should be generated at docs-build-time.

File: .readthedocs.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  tools:
+    python: "3.12"
+  commands:
+    - asdf plugin add uv
+    - asdf install uv latest
+    - asdf global uv latest
+    - uv venv
+    - uv pip install .[docs]
+    # Generate api docs
+    - sphinx-apidoc -o $READTHEDOCS_OUTPUT/docs/api/ --no-toc --module-first
```
Comment (trey-stafford) [inline .readthedocs.yaml:17]: Everything above this is from the scientific python template. These commands reflect the guide & nox task command provided in the template.

Currently using invoke, but I think we could switch back to nox :) That can be a separate PR.

There's an invoke task that runs these same commands.

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+:hidden:
+
+```
+
+```{toctree}
+:maxdepth: 2
+:hidden:
+:caption: API
+
+api/*
```
Comment (trey-stafford) [inline docs/index.md:14]: TODO: not sure this is working as expected? Or at all? I'd like to have a TOC on the side that has a link to the api docs...

File: tasks/docs.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from invoke import task
+
+from .util import PROJECT_DIR, print_and_run
+
+
+@task()
+def build(_ctx):
```
Comment (trey-stafford) [inline tasks/docs.py:9]: a `watch` task would also be nice for when docs are being actively worked on.

File: .gitignore (hunk 1 above)
Comment (mfisher87) [reply .gitignore:3]: :100:

File: .readthedocs.yaml (hunk 2 above)
Comment (mfisher87) [reply .readthedocs.yaml:17]: >Currently using invoke, but I think we could switch back to nox

Curious what your thoughts are! I think we may have already talked about why you switched to invoke, but I don't remember what you said :X

File: tasks/docs.py (hunk 4 above)
Comment (mfisher87) [reply tasks/docs.py:9]: :100:

File: .readthedocs.yaml (hunk 2 above)
Comment (trey-stafford) [reply .readthedocs.yaml:17]: I think the main complication was getting it to work with my conda environment. I needed to set the PYTHONPATH to the src manually, and doing that borked the nox environment somehow. I tried a few different ways but never got it to work. 

I think now that we don't really rely on conda, and that we can do a `pip install -e`, the manual PYTHONPATH setting is no longer necessary & nox should just work. Hopefully.

Comment (mfisher87) [reply .readthedocs.yaml:17]: That all makes sense, thanks!

File: tasks/docs.py (hunk 4 above)
Comment (trey-stafford) [reply tasks/docs.py:22]: Added this and a `clean` task as well.
